sycl/include/sycl/builtins.hpp

```diff
@@ -789,6 +896,79 @@ detail::enable_if_t<detail::is_svgenfloat<T>::value, T> sign(T x) __NOEXC {
   return __sycl_std::__invoke_sign<T>(x);
 }
 
+// marray common functions
+
+// TODO: can be optimized in the way math functions are optimized (usage of
+// vec<T, 2>)
+#define __SYCL_MARRAY_COMMON_FUNCTION_UNOP_OVERLOAD(NAME, ARGTYPE, ARGVAL,     \
+                                                    ITEM)                      \
+  template <typename T,                                                        \
+            typename = std::enable_if_t<detail::is_mgenfloat<T>::value>>       \
+  sycl::marray<detail::marray_element_type<T>, T::size()> NAME(ARGTYPE ARGVAL) \
+      __NOEXC {                                                                \
+    sycl::marray<detail::marray_element_type<T>, T::size()> res;               \
+    for (int i = 0; i < ARGVAL.size(); i++) {                                  \
```

Comment:
We could use `T::size()` here and replace `ARGTYPE ARGVAL` with a simple `ARG` to align this macro definition with `BINOP` and `TEROP`